### PR TITLE
CloudOps: Who has this IP?

### PR DESCRIPTION
**Pull request by iri989**

### CODE REVIEW

File: cosmic-client/src/main/webapp/scripts/cloudOps.js [hunk 1]
```diff
@@ -139,7 +142,75 @@
...
+                        ipstate: {
+                            label: 'label.state'
+                        },
+                        created: {
+                            label:'label.created',
+                            converter: cloudStack.converters.toLocalDate
+                        }
+                    },
+                    dataProvider: function (args) {
+                        console.log(args.filterBy.search.value);
```
Comment (eliasgomes) [inline cosmic-client/src/main/webapp/scripts/cloudOps.js]: Please remove the `console.log` instruction.

File: cosmic-core/api/src/main/java/com/cloud/api/response/WhoHasThisIpResponse.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+    @SerializedName(ApiConstants.NETWORK_NAME)
+    @Param(description = "Network name")
+    private String networkname;
+
+    @SerializedName(ApiConstants.ASSOCIATED_NETWORK_NAME)
+    @Param(description = "Network name")
+    private String associatedNetworkName;
+
+    @SerializedName(ApiConstants.PRIVATE_MAC_ADDRESS)
```
Comment (eliasgomes) [inline cosmic-core/api/src/main/java/com/cloud/api/response/WhoHasThisIpResponse.java]: You created `ApiConstants.NICS_MAC_ADDRESS` but are using `ApiConstants.PRIVATE_MAC_ADDRESS`. An autocompletion mistake probably?

NOTE: Why not just `ApiConstants.MAC_ADDRESS`?

File: cosmic-client/src/main/webapp/l10n/en.js [hunk 3]
```diff
@@ -1677,6 +1678,7 @@ var dictionary = {
     "label.welcome": "Welcome",
     "label.welcome.cloud.console": "Welcome to Management Console",
     "label.what.is.cloudstack": "What is Cosmic?",
+    "label.whohasthisip": "Who has this IP",
```
Comment (eliasgomes) [inline cosmic-client/src/main/webapp/l10n/en.js]: Who has this IP? (with a question mark)

File: cosmic-client/src/main/webapp/scripts/cloudOps.js [hunk 4]
```diff
@@ -139,7 +142,75 @@
...
+                            success: function (json) {
+                                whohasthisip = json.listwhohasthisipresponse.whohasthisip ? json.listwhohasthisipresponse.whohasthisip : [];
+                                args.response.success({
+                                    data: $.map(whohasthisip, function (whohasthisip) {
+                                        return {
+                                            id: whohasthisip.id,
+                                            virtualmachine: whohasthisip.virtualmachinename,
+                                            networkname: whohasthisip.networkname,
+                                            associatednetwork: whohasthisip.associatednetworkname,
+                                            macaddress: whohasthisip.privatemacaddress,
```
Comment (eliasgomes) [inline cosmic-client/src/main/webapp/scripts/cloudOps.js]: Please see my other comment about the MAC address.

File: cosmic-core/api/src/main/java/com/cloud/api/response/WhoHasThisIpResponse.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+import com.cloud.serializer.Param;
+import com.cloud.utils.net.Ip;
+import com.cloud.vm.Nic;
+
+import java.net.URI;
+import java.util.Date;
+
+import com.google.gson.annotations.SerializedName;
+
+public class WhoHasThisIpResponse extends BaseResponse {
```
Comment (eliasgomes) [inline cosmic-core/api/src/main/java/com/cloud/api/response/WhoHasThisIpResponse.java:14]: I think it's a good idea to also include the UUID next to the name of the resources that are being listed in the response. Something that could be shown only in the detailed view for example. That way we can avoid people looking up these UUIDs in the database.

**Comment by remibergsma** [comment]
go build

**Pull request merged**